dali_tf_plugin/setup.py.in

```diff
@@ -14,16 +14,64 @@
 
 from setuptools import setup, find_namespace_packages
 from dali_tf_plugin_install_tool import InstallerHelper
-from setuptools.command.build_ext import build_ext
 from setuptools.dist import Distribution
 from setuptools.extension import Extension
 import os
+import sys
+
+from setuptools.command.build_ext import build_ext
+from setuptools.command.egg_info import egg_info
+
+DALI_PKG_REQ = 'nvidia-dali@DALI_FLAVOR_MINUS@-cuda@CUDA_VERSION_SHORT_DIGIT_ONLY@==@DALI_VERSION@'
+NVIDIA_INDEX_URL = "https://developer.download.nvidia.com/compute/redist/@DALI_FLAVOR@"
+
+def is_dali_installed():
+    try:
+        import nvidia.dali
+        return True
+    except:
+        return False
+
+def pip_install(pkg_req, user=False):
+    pip_args = [
+        "install",
+        "--index-url", NVIDIA_INDEX_URL
+    ]
+    if user:
+        pip_args.append("--user")
+    pip_args.append(pkg_req)
+
+    from pip._internal import main as pip
```

Comment:
`__internal ` doesn't sound like a public PIP api. https://github.com/NVIDIA/DALI/blob/master/qa/setup_packages.py#L7L13 shows that `_internal` can be changed version to version

Reply:
Yes, I was worried about that too. Another option is to invoke pip program directly